sound/soc/sof/intel/hda-loader.c

```diff
@@ -44,7 +44,7 @@ static int cl_stream_prepare(struct snd_sof_dev *sdev, unsigned int format,
 	int ret;
 
 	if (direction == SNDRV_PCM_STREAM_PLAYBACK) {
-		stream = hda_dsp_stream_get_pstream(sdev);
+		stream = hda_dsp_stream_get(sdev, direction);
```

Comment:
this doesn't seem right, in one case the second argument is direction and in the other it's a stream.

Reply:
for substream->stream, the definition in kernel is:
int stream; /* stream (direction) */
so it is direction and not confict